src/context.rs

```diff
@@ -21,6 +21,49 @@ use crate::smtp::*;
 use crate::types::*;
 use crate::x::*;
 
+const CONFIG_KEYS: [&'static str; 33] = [
```

Comment:
Would it not be tempting to make this an enum for the rust code, only needing the runtime-checked string version for the C API?